3 Module Development Exercise - Recursive Unhider/recursiveUnhider-rt/src/com/tridiumuniversity/recursiveUnhider/BRecursiveUnhider.java

```diff
@@ -0,0 +1,102 @@
+package com.tridiumuniversity.recursiveUnhider;
+
+import javax.baja.nre.annotations.NiagaraAction;
+import javax.baja.nre.annotations.NiagaraProperty;
+import javax.baja.nre.annotations.NiagaraType;
+import javax.baja.sys.*;
+import java.util.logging.Logger;
+import java.util.stream.Stream;
+
+@NiagaraType
+@NiagaraProperty(name = "regex", type = "String", defaultValue = "")
+@NiagaraAction(name = "unhide", flags = Flags.ASYNC)
```

Comment:
I like the insight that this is something that might be a good idea to do asynchronously. We'll talk more about this in class today, but the `ASYNC` flag is something that you probably shouldn't use unless you also override `BComponent#post`. Basically this does invoke the action on another thread, but it puts the action invocation on a critical thread in Niagara called the engine thread, which we want to avoid.

Reply:
Good to know, thanks for the feedback!